persistence/trees/trees_test.go

```diff
@@ -1,19 +1,216 @@
+		assert.NoError(t, err)
+		assert.NotEmpty(t, hash)
+	})
+}
+
+func newTestPersistenceModule(t *testing.T, databaseUrl string) modules.PersistenceModule {
+	t.Helper()
+	teardownDeterministicKeygen := keygen.GetInstance().SetSeed(42)
+	defer teardownDeterministicKeygen()
+
+	cfg := &configs.Config{
+		Persistence: &configs.PersistenceConfig{
+			PostgresUrl:       databaseUrl,
+			NodeSchema:        testSchema,
+			BlockStorePath:    ":memory:",
+			TxIndexerPath:     ":memory:",
+			TreesStoreDir:     ":memory:",
+			MaxConnsCount:     5,
+			MinConnsCount:     1,
+			MaxConnLifetime:   "5m",
+			MaxConnIdleTime:   "1m",
+			HealthCheckPeriod: "30s",
+		},
+	}
+
+	genesisState, _ := test_artifacts.NewGenesisState(
+		genesisStateNumValidators,
+		genesisStateNumServicers,
+		genesisStateNumApplications,
+		genesisStateNumServicers,
+	)
+
+	runtimeMgr := runtime.NewManager(cfg, genesisState)
+
+	bus, err := runtime.CreateBus(runtimeMgr)
+	assert.NoError(t, err)
+
+	persistenceMod, err := persistence.Create(bus)
+	assert.NoError(t, err)
+
+	return persistenceMod.(modules.PersistenceModule)
 }
+func createAndInsertDefaultTestApp(t *testing.T, db *persistence.PostgresContext) (*coreTypes.Actor, error) {
+	app := newTestApp(t)
+
+	addrBz, err := hex.DecodeString(app.Address)
+	require.NoError(t, err)
+
+	pubKeyBz, err := hex.DecodeString(app.PublicKey)
+	require.NoError(t, err)
+
+	outputBz, err := hex.DecodeString(app.Output)
+	require.NoError(t, err)
+	return app, db.InsertApp(
+		addrBz,
+		pubKeyBz,
+		outputBz,
+		false,
+		DefaultStakeStatus,
+		DefaultStake,
+		DefaultChains,
+		DefaultPauseHeight,
+		DefaultUnstakingHeight)
+}
+
+func newTestApp(t *testing.T) *coreTypes.Actor {
+	operatorKey, err := crypto.GeneratePublicKey()
+	require.NoError(t, err)
+
+	outputAddr, err := crypto.GenerateAddress()
+	require.NoError(t, err)
+
+	return &coreTypes.Actor{
+		Address:         hex.EncodeToString(operatorKey.Address()),
+		PublicKey:       hex.EncodeToString(operatorKey.Bytes()),
+		Chains:          DefaultChains,
+		StakedAmount:    DefaultStake,
+		PausedHeight:    DefaultPauseHeight,
+		UnstakingHeight: DefaultUnstakingHeight,
+		Output:          hex.EncodeToString(outputAddr),
+	}
+}
+
+func NewTestPostgresContext(t testing.TB, height int64, testPersistenceMod modules.PersistenceModule) *persistence.PostgresContext {
+	rwCtx, err := testPersistenceMod.NewRWContext(height)
+	if err != nil {
+		log.Fatalf("Error creating new context: %v\n", err)
+	}
+
+	postgresCtx, ok := rwCtx.(*persistence.PostgresContext)
+	if !ok {
+		log.Fatalf("Error casting RW context to Postgres context")
+	}
+
+	// TECHDEBT: This should not be part of `NewTestPostgresContext`. It causes unnecessary resets
+	// if we call `NewTestPostgresContext` more than once in a single test.
+	t.Cleanup(func() {
+		resetStateToGenesis(testPersistenceMod)
+	})
+
+	return postgresCtx
+}
+
+func NewTestPersistenceModule(t *testing.T, databaseUrl string) modules.PersistenceModule {
```

Comment:
Unsure if this is in progress, but can we add a `TECHDEBT` or `TODO` comment on some of these functions to move them into `internal` for global use? I have copies of these functions in IBC pretty sure as there aren't any globally accessable ones.

Reply:
This is "in progress" in that it's being discussed, but what you're describing is similar to what we've identified in the [TestUtils](https://www.notion.so/pocketnetwork/Testutils-9cba9010e18447248e9daa8a3b87e3f2#f1d7f404918541bc86ccde07980e3f09) document. I'll add a TECHDEBT comment here, though. This is definitely a function that would benefit from being in a clearly visible & central place for testing.